src/extension/wps/src/main/java/au/org/emii/aggregator/variable/UnpackerOverrides.java

```diff
@@ -0,0 +1,114 @@
+package au.org.emii.aggregator.variable;
+
+import ucar.ma2.DataType;
+
+import java.util.Arrays;
+
+/**
+ * Optional overrides for values to be used in the unpacked variable
+ */
+public class UnpackerOverrides {
+    private final Number newFillerValue;
+    private final Number[] newValidRange;
+    private final Number newValidMin;
+    private final Number newValidMax;
+    private final Number[] newMissingValues;
+    private final DataType newDataType;
+
+    private UnpackerOverrides(Builder builder) {
+        this.newFillerValue = builder.newFillerValue;
+        this.newValidRange = builder.newValidRange == null ? null :
+            Arrays.copyOf(builder.newValidRange, builder.newValidRange.length);
+        this.newValidMin = builder.newValidMin;
+        this.newValidMax = builder.newValidMax;
+        this.newMissingValues = builder.newMissingValues == null ? null :
+            Arrays.copyOf(builder.newMissingValues, builder.newMissingValues.length);
+        this.newDataType = builder.newDataType;
+    }
+
+    public Number getNewFillerValue() {
+        return newFillerValue;
+    }
+
+    public Number[] getNewValidRange() {
+        return newValidRange == null ? null :Arrays.copyOf(newValidRange, newValidRange.length);
+    }
+
+    public Number getNewValidMin() {
+        return newValidMin;
+    }
+
+    public Number getNewValidMax() {
+        return newValidMax;
+    }
+
+    public Number[] getNewMissingValues() {
+        return newMissingValues == null ? null : Arrays.copyOf(newMissingValues, newMissingValues.length);
+    }
+
+    public DataType getNewDataType() {
+        return newDataType;
+    }
+
+    public static class Builder {
+        private Number newFillerValue;
+        private Number[] newValidRange;
+        private Number newValidMin;
+        private Number newValidMax;
+        private Number[] newMissingValues;
+        private DataType newDataType;
+
+        public Builder newFillerValue(Number newFillerValue) {
+            this.newFillerValue = newFillerValue;
+            return this;
+        }
+
+        public Builder newValidRange(Number[] newValidRange) {
```

Comment:
If range is limited to two values, then min max variables might be better than using an array.